web3.js/src/fetch-impl.ts

```diff
@@ -0,0 +1,13 @@
+import * as nodeFetch from 'node-fetch';
+
+export * from 'node-fetch';
+export default async function (
+  input: nodeFetch.RequestInfo,
+  init?: nodeFetch.RequestInit,
+): Promise<nodeFetch.Response> {
+  const processedInput =
+    typeof input === 'string' && input.slice(0, 2) === '//'
```

Comment:
when would the input be missing a protocol and just start with `//`?

Reply:
This is a standard patch for the broken implementation of `node-fetch` that doesn't support protocol-relative URLs like the `fetch` in browsers does. https://github.com/node-fetch/node-fetch/blob/main/docs/v2-LIMITS.md

Folks can and will write apps that make use of protocol-relative URLs, and we don't want them to break in Node when they do.

This replicates what `cross-fetch` was already doing.

https://github.com/lquixada/cross-fetch/blob/main/src/node-ponyfill.js#L4-L11